src/main/scala/sam/sceval/BinUtils.scala

```diff
@@ -6,8 +6,11 @@ object BinUtils {
   def binnerFac[Model](partitionLastIndexes: Array[Map[Model, Long]],
                        numRecodsPerBin: Long): (Model, Long, Int) => Int = {
     val modelToBinStats: Map[Model, Array[BinStats]] =
+        // GMARIO: what's wrong with map and reduce instead of fold?
       partitionLastIndexes.flatMap(_.keySet).toSet.foldLeft(Map.empty[Model, List[BinStats]])((modelToStats, model) =>
         partitionLastIndexes.foldLeft(modelToStats)((modelToStats, partition) =>
+          // GMARIO: you are doing getOrElse and then match, you could do get and match the option directly.
```

Comment:
Actually, I'm matching on the result, if I tried to use the option directly I would need a nested match, therefore the way it is is clearer.